### ISSUE TITLE
Weird errors when installing with npm

### ISSUE THREAD

**Issue by ughstudios** [Gielert/NoodleJS#21]
What is this??

PS C:\Users\decil\Desktop\HVS_Game\Content\Scripts> npm install noodle.js
npm WARN deprecated node-opus@0.3.3: This project is unmaintained. See @discordjs/opus for an alternative.

> ref@1.3.5 install C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref
> node-gyp rebuild

C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  binding.cc
  win_delay_load_hook.cc
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(222,43): error C2661: 'v8::Value::Boole
anValue': no overloaded function takes 0 arguments [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\bu
ild\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(253,38): error C2660: 'v8::Value::Uint3
2Value': function does not take 0 arguments [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\bin
ding.vcxproj]
C:\Users\decil\AppData\Local\node-gyp\Cache\12.16.3\include\node\v8.h(2707,41): message : see declaration of 'v8::Value
::Uint32Value' (compiling source file ..\src\binding.cc) [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\
ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(360,31): error C2664: 'v8::String::Utf8
Value::Utf8Value(const v8::String::Utf8Value &)': cannot convert argument 1 from 'v8::Local<v8::Value>' to 'const v8::S
tring::Utf8Value &' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(360,31): message : Reason: cannot conve
rt from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_m
odules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(360,28): message : No user-defined-conv
ersion operator available that can perform this conversion, or the operator cannot be called [C:\Users\decil\Desktop\HV
S_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\AppData\Local\node-gyp\Cache\12.16.3\include\node\v8.h(3142,5): message : see declaration of 'v8::String
::Utf8Value::Utf8Value' (compiling source file ..\src\binding.cc) [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node
_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(447,31): error C2664: 'v8::String::Utf8
Value::Utf8Value(const v8::String::Utf8Value &)': cannot convert argument 1 from 'v8::Local<v8::Value>' to 'const v8::S
tring::Utf8Value &' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(447,31): message : Reason: cannot conve
rt from 'v8::Local<v8::Value>' to 'const v8::String::Utf8Value' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_m
odules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(447,28): message : No user-defined-conv
ersion operator available that can perform this conversion, or the operator cannot be called [C:\Users\decil\Desktop\HV
S_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\AppData\Local\node-gyp\Cache\12.16.3\include\node\v8.h(3142,5): message : see declaration of 'v8::String
::Utf8Value::Utf8Value' (compiling source file ..\src\binding.cc) [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node
_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(521,38): error C2660: 'v8::Value::Uint3
2Value': function does not take 0 arguments [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\bin
ding.vcxproj]
C:\Users\decil\AppData\Local\node-gyp\Cache\12.16.3\include\node\v8.h(2707,41): message : see declaration of 'v8::Value
::Uint32Value' (compiling source file ..\src\binding.cc) [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\
ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(550,44): error C2660: 'v8::Value::Uint3
2Value': function does not take 0 arguments [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\bin
ding.vcxproj]
C:\Users\decil\AppData\Local\node-gyp\Cache\12.16.3\include\node\v8.h(2707,41): message : see declaration of 'v8::Value
::Uint32Value' (compiling source file ..\src\binding.cc) [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\
ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(582,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(583,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(584,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(585,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(586,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(587,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(588,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(589,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(590,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(591,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(593,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(594,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(595,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(596,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(597,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(598,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(599,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(600,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(601,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(602,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(603,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(604,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(605,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(606,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(608,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(615,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(616,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(617,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(618,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(619,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(620,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(621,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(622,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(623,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(624,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(625,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(626,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(627,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(628,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(629,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(630,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(631,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(632,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(633,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(634,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(635,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(636,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(637,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(638,3): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(641,9): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(642,9): warning C4996: 'v8::Object::Set
': was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(643,8): warning C4996: 'Nan::ForceSet':
 was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\src\binding.cc(644,8): warning C4996: 'Nan::ForceSet':
 was declared deprecated [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref\build\binding.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Windows_NT 10.0.19041
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\ref
gyp ERR! node -v v12.16.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok

> node-opus@0.3.3 install C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus
> node-gyp rebuild

C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
  opus.c
  opus_decoder.c
  opus_encoder.c
  opus_multistream.c
  opus_multistream_encoder.c
  opus_multistream_decoder.c
  repacketizer.c
  analysis.c
  mlp.c
  mlp_data.c
  bands.c
  celt.c
  celt_encoder.c
  celt_decoder.c
  cwrs.c
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\deps\opus\src\opus_encoder.c(655,1): warning C43
34: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?) [C:\Users\decil\Desktop\H
VS_Game\Content\Scripts\node_modules\node-opus\build\deps\libopus.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\deps\opus\src\opus_encoder.c(728,1): warning C43
34: '<<': result of 32-bit shift implicitly converted to 64 bits (was 64-bit shift intended?) [C:\Users\decil\Desktop\H
VS_Game\Content\Scripts\node_modules\node-opus\build\deps\libopus.vcxproj]
  entcode.c
  entdec.c
  entenc.c
  kiss_fft.c
  laplace.c
  mathops.c
  mdct.c
  modes.c
  pitch.c
  celt_lpc.c
  quant_bands.c
  rate.c
  vq.c
  CNG.c
  code_signs.c
  init_decoder.c
  decode_core.c
  decode_frame.c
  decode_parameters.c
  decode_indices.c
  decode_pulses.c
  decoder_set_fs.c
  dec_API.c
  enc_API.c
  encode_indices.c
  encode_pulses.c
  gain_quant.c
  interpolate.c
  LP_variable_cutoff.c
  NLSF_decode.c
  NSQ.c
  NSQ_del_dec.c
  PLC.c
  shell_coder.c
  tables_gain.c
  tables_LTP.c
  tables_NLSF_CB_NB_MB.c
  tables_NLSF_CB_WB.c
  tables_other.c
  tables_pitch_lag.c
  tables_pulses_per_block.c
  VAD.c
  control_audio_bandwidth.c
  quant_LTP_gains.c
  VQ_WMat_EC.c
  HP_variable_cutoff.c
  NLSF_encode.c
  NLSF_VQ.c
  NLSF_unpack.c
  NLSF_del_dec_quant.c
  process_NLSFs.c
  stereo_LR_to_MS.c
  stereo_MS_to_LR.c
  check_control_input.c
  control_SNR.c
  init_encoder.c
  control_codec.c
  A2NLSF.c
  ana_filt_bank_1.c
  biquad_alt.c
  bwexpander_32.c
  bwexpander.c
  debug.c
  decode_pitch.c
  inner_prod_aligned.c
  lin2log.c
  log2lin.c
  LPC_analysis_filter.c
  LPC_inv_pred_gain.c
  table_LSF_cos.c
  NLSF2A.c
  NLSF_stabilize.c
  NLSF_VQ_weights_laroia.c
  pitch_est_tables.c
  resampler.c
  resampler_down2_3.c
  resampler_down2.c
  resampler_private_AR2.c
  resampler_private_down_FIR.c
  resampler_private_IIR_FIR.c
  resampler_private_up2_HQ.c
  resampler_rom.c
  sigm_Q15.c
  sort.c
  sum_sqr_shift.c
  stereo_decode_pred.c
  stereo_encode_pred.c
  stereo_find_predictor.c
  stereo_quant_pred.c
  apply_sine_window_FLP.c
  corrMatrix_FLP.c
  encode_frame_FLP.c
  find_LPC_FLP.c
  find_LTP_FLP.c
  find_pitch_lags_FLP.c
  find_pred_coefs_FLP.c
  LPC_analysis_filter_FLP.c
  LTP_analysis_filter_FLP.c
  LTP_scale_ctrl_FLP.c
  noise_shape_analysis_FLP.c
  prefilter_FLP.c
  process_gains_FLP.c
  regularize_correlations_FLP.c
  residual_energy_FLP.c
  solve_LS_FLP.c
  warped_autocorrelation_FLP.c
  wrappers_FLP.c
  autocorrelation_FLP.c
  burg_modified_FLP.c
  bwexpander_FLP.c
  energy_FLP.c
  inner_product_FLP.c
  k2a_FLP.c
  levinsondurbin_FLP.c
  LPC_inv_pred_gain_FLP.c
  pitch_analysis_core_FLP.c
  scale_copy_vector_FLP.c
  scale_vector_FLP.c
  schur_FLP.c
  sort_FLP.c
  win_delay_load_hook.cc
  libopus.vcxproj -> C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\build\Release\\libopus.lib
  node-opus.cc
  win_delay_load_hook.cc
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\src\node-opus.cc(41,19): warning C4099: 'OpusEnc
oder': type name first seen using 'struct' now seen using 'class' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node
_modules\node-opus\build\node-opus.vcxproj]
C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\src\node-opus.cc(41): message : see declaration
of 'OpusEncoder' [C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\build\node-opus.vcxproj]
     Creating library C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\build\Release\node-opus.li
  b and object C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\build\Release\node-opus.exp
  node-opus.vcxproj -> C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\node-opus\build\Release\\node-opus.
  node

> protobufjs@6.9.0 postinstall C:\Users\decil\Desktop\HVS_Game\Content\Scripts\node_modules\protobufjs
> node scripts/postinstall

npm WARN scripts@1.0.0 No description
npm WARN scripts@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: ref@1.3.5 (node_modules\ref):
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: ref@1.3.5 install: `node-gyp rebuild`
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: Exit status 1

+ noodle.js@1.3.0
added 26 packages from 64 contributors and audited 29 packages in 21.188s
found 0 vulnerabilities

**Comment by jsjb** [comment]
Looks like [node-opus](https://github.com/Rantanen/node-opus) finally bit the big one last year - announced in discordjs/discord.js#3678 and Rantanen/node-opus#102. It depends on `ref`, de facto deprecated for Node 12+ per TooTallNate/ref#125. Debian stable (Buster) is still maintaining Node 10 (https://packages.debian.org/buster/nodejs), so this issue wouldn't be seen by people running NoodleJS bots on some Linux servers.

That dependency causes the node-gyp build errors you see (also visible at the end of nodejs/node-gyp#1754). Workaround is target, build, and deploy only on Debian Stable :p

For real though, it should be no problem to migrate to [discordjs/opus](https://github.com/discordjs/opus) as recommended in the node-opus README - the API is nearly the same. I'll make time after work to put a PR together.

**Comment by ughstudios** [comment]
> 
> 
> Looks like [node-opus](https://github.com/Rantanen/node-opus) finally bit the big one last year - announced in [discordjs/discord.js#3678](https://github.com/discordjs/discord.js/issues/3678) and [Rantanen/node-opus#102](https://github.com/Rantanen/node-opus/issues/102). It depends on `ref`, de facto deprecated for Node 12+ per [TooTallNate/ref#125](https://github.com/TooTallNate/ref/issues/125). Debian stable (Buster) is still maintaining Node 10 (https://packages.debian.org/buster/nodejs), so this issue wouldn't be seen by people running NoodleJS bots on some Linux servers.
> 
> That dependency causes the node-gyp build errors you see (also visible at the end of [nodejs/node-gyp#1754](https://github.com/nodejs/node-gyp/issues/1754)). Workaround is target, build, and deploy only on Debian Stable :p
> 
> For real though, it should be no problem to migrate to [discordjs/opus](https://github.com/discordjs/opus) as recommended in the node-opus README - the API is nearly the same. I'll make time after work to put a PR together.

Hey buddy, I'm trying to use Mumble to integrate with my unreal engine 4 game that I am building. I essentially need to take any audio data after a player connects to a channel, and send it to an audio component (binary arrays would work fine).

**Comment by ughstudios** [comment]
> 
> 
> Looks like [node-opus](https://github.com/Rantanen/node-opus) finally bit the big one last year - announced in [discordjs/discord.js#3678](https://github.com/discordjs/discord.js/issues/3678) and [Rantanen/node-opus#102](https://github.com/Rantanen/node-opus/issues/102). It depends on `ref`, de facto deprecated for Node 12+ per [TooTallNate/ref#125](https://github.com/TooTallNate/ref/issues/125). Debian stable (Buster) is still maintaining Node 10 (https://packages.debian.org/buster/nodejs), so this issue wouldn't be seen by people running NoodleJS bots on some Linux servers.
> 
> That dependency causes the node-gyp build errors you see (also visible at the end of [nodejs/node-gyp#1754](https://github.com/nodejs/node-gyp/issues/1754)). Workaround is target, build, and deploy only on Debian Stable :p
> 
> For real though, it should be no problem to migrate to [discordjs/opus](https://github.com/discordjs/opus) as recommended in the node-opus README - the API is nearly the same. I'll make time after work to put a PR together.

Did you ever make the pull request to fix the issue? Will this work on the latest version of Node?

**Comment by jsjb** [comment]
I forked the repo and pushed some code to my GitHub that builds on modern Node, but I was also looking at what I think is a bug in the original code. For some reason, DispatcherStream emits the 'finish' event while data is still being written to the stream, so tests fail on newer Mocha, which waits for cleanup of sockets and things like that. For the sake of pushing code at all, I've just passed `--exit` to mocha to replicate the old behaviour and make the tests pass "*for now*".

Regarding the project you want this for, NoodleJS actually doesn't currently support processing received audio data (#17). It handles everything up to that point - which is clearly marked with a helpful TODO comment  ([src/Connection.js#L59](https://github.com/Gielert/NoodleJS/blob/d67ceec7ee1c605265d9a05a3ea99876fedd000c/src/Connection.js#L59)) - so support wouldn't be too hard to finish.
